apps/src/netsim/NetSimTable.js

```diff
@@ -73,10 +73,10 @@ var NetSimTable = module.exports = function (storageTable) {
  */
 NetSimTable.prototype.readAll = function (callback) {
   this.remoteTable_.readAll(function (err, data) {
-    callback(err, data);
     if (err === null) {
       this.fullCacheUpdate_(data);
     }
+    callback(err, data);
```

Comment:
@breville suggested this reversal a while back, and at the time I thought I didn't want to change things because I thought my way was a better reflection of how the code would have to respond to remote events.

That was totally garbage.  Turns out I've been using and optimizing for immediate notification of local changes anyway.  I lean on this 'callback-at-the-end' pattern more than I realize, and it's just caused pain.  So here I'm switching over to the more conventional form, and lo! not a single test broke, everything still seems to work great.

Thanks Brendan :bow: